jena-arq/src/main/java/org/apache/jena/sparql/algebra/optimize/Optimize.java

```diff
@@ -18,82 +18,58 @@
 
 package org.apache.jena.sparql.algebra.optimize;
 
-import org.apache.jena.query.ARQ ;
 import org.apache.jena.sparql.ARQConstants ;
-import org.apache.jena.sparql.SystemARQ ;
-import org.apache.jena.sparql.algebra.* ;
-import org.apache.jena.sparql.algebra.op.OpLabel ;
+import org.apache.jena.sparql.algebra.Op ;
 import org.apache.jena.sparql.engine.ExecutionContext ;
 import org.apache.jena.sparql.util.Context ;
-import org.apache.jena.sparql.util.Symbol ;
-import org.slf4j.Logger ;
-import org.slf4j.LoggerFactory ;
 
-
-public class Optimize implements Rewrite
+/** Optimization of algebra expressions. 
+ * <p>
+ *  New optimization processes can be installed via a global change:
+ *  <pre>
+ *    Optimize.setFactory((cxt)->new MyOptimize(cxt)) ;</pre>
+ *  or on a per-context basis: 
+ *  <pre>
+ *    Optimize.RewriterFactory f = (cxt)->new MyOptimize(cxt) ;
+ *    context.set(ARQConstants.sysOptimizerFactory, f) ;<pre>
+ */
+public class Optimize
 {
-    static private Logger log = LoggerFactory.getLogger(Optimize.class) ;
-
-    // A small (one slot) registry to allow plugging in an alternative optimizer
+    /** Factory for rewriters */
```

Comment:
What difference does it make? It's an interface; there is no static/enclosing distinction.